File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int,int> um;
        for(auto x: nums){
            if(um.find(x)!=um.end()){
                return true;
            }else{
                um[x]++;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        int[] ans = new int[k];
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int n:nums){
            map.put(n,map.getOrDefault(n,0)+1);
        }
        List<Map.Entry<Integer,Integer>> list= new ArrayList<>(map.entrySet());
        Collections.sort(list, (a,b)-> b.getValue()-a.getValue());
        for(int i=0;i<k;i++){
            ans[i] = list.get(i).getKey();
        }
        return ans;
    }

}
